engine/src/flutter/shell/platform/embedder/embedder_engine.cc

```diff
@@ -243,8 +243,19 @@ bool EmbedderEngine::RunTask(const FlutterTask* task) {
   if (task == nullptr) {
     return false;
   }
-  return thread_host_->PostTask(reinterpret_cast<int64_t>(task->runner),
-                                task->task);
+  auto result = thread_host_->PostTask(reinterpret_cast<int64_t>(task->runner),
+                                       task->task);
+  // Normally the microtask queue is flushed through MessageLoopTaskQueues
+  // observer. When running with application specified UI task runner
+  // which is not associated with a task queue, the microtask queue needs to
+  // be flushed manually after running the task.
```

Comment:
> Normally the microtask queue...

Once we merge UI/platform threads on all platform, will this case become the "abnormal" case? Or is this the normal case for iOS/Android with merged threads?

What do you think of a comment like this?

```suggestion
  // If the UI and platform threads are separate, the microtask queue is
  // flushed through MessageLoopTaskQueues observer.
  // If the UI and platform threads are merged, the UI task runner has no
  // associated task queue, and microtasks need to be flushed manually
  // after running the task.
```